src/reset/index.ts

```diff
@@ -1,15 +1,21 @@
-import { Unit, Store } from 'effector';
+import { createEvent } from 'effector';
 
-export function reset({
-  clock,
-  target,
-}: {
-  clock: Unit<any> | Array<Unit<any>>;
+import type { Event, Unit, Store } from 'effector';
+
+type Params = {
+  clock?: Unit<any> | Array<Unit<any>>;
   target: Store<any> | Array<Store<any>>;
-}): void {
+};
+
+export function reset(_: Required<Params>): void;
+export function reset(_: Pick<Params, 'target'>): Event<void>;
+
+export function reset({ clock, target }: Params) {
   const targets = Array.isArray(target) ? target : [target];
-  const clocks = Array.isArray(clock) ? clock : [clock];
+  const clocks = Array.isArray(clock) ? clock : [clock ?? createEvent()];
   targets.forEach((target) => {
     target.reset(clocks);
   });
+
+  return clock === undefined ? clocks[0] : undefined;
```

Comment:
```suggestion
  return clocks[0];
```

Reply:
I thought that if the original сlock is passed, we do not return it 🤔 
We'll always return it?

Reply:
I think it would be simpler, also when we plan to always return Event

Reply:
U're right

Reply:
Still, I don't think we should do it that way, and here's why:

- if clock is passed, it can be Unit | Unit[]
- so we have to return an array
- If we return an array when we didn't pass clock, we have to apply destructuring, which is not good for DX
- If we return Unit[] | Unit | Event, the return type is not consistent

Reply:
Hm, makes sense

Reply:
What about to infer unit type and return is as is, only if passed a single unit?

Reply:
I propose to return the Event only if the clock has not been transferred